jfrog-oauth/run.sh

```diff
@@ -47,12 +53,75 @@ index-url = https://${ARTIFACTORY_USERNAME}:${ARTIFACTORY_ACCESS_TOKEN}@${JFROG_
 EOF
 fi
 
-# Set GOPROXY to use the Artifactory "go" repository.
+# Configure Artifactory "go" repository.
 if [ -z "${REPOSITORY_GO}" ]; then
-  echo "🤔 REPOSITORY_GO is not set, skipping go configuration."
+  echo "🤔 no go repository is set, skipping go configuration."
+  echo "You can configure a go repository by providing the a key for 'go' in the 'package_managers' input."
 else
   echo "🐹 Configuring go..."
-  jf go-config --global --repo-resolve "${REPOSITORY_GO}"
-  export GOPROXY="https://${ARTIFACTORY_USERNAME}:${ARTIFACTORY_ACCESS_TOKEN}@${JFROG_HOST}/artifactory/api/go/${REPOSITORY_GO}"
+  jf goc --global --repo-resolve "${REPOSITORY_GO}"
 fi
 echo "🥳 Configuration complete!"
+
+# Configure the JFrog CLI to use the Artifactory "docker" repository.
+if [ -z "${REPOSITORY_DOCKER}" ]; then
+  echo "🤔 no docker repository is set, skipping docker configuration."
+  echo "You can configure a docker repository by providing the a key for 'docker' in the 'package_managers' input."
+else
+  echo "🐳 Configuring docker..."
+  mkdir -p ~/.docker
+  cat << EOF > ~/.docker/config.json
+{
+  "auths": {
+    "${JFROG_HOST}": {
+      "auth": "$(echo -n "${ARTIFACTORY_USERNAME}:${ARTIFACTORY_ACCESS_TOKEN}" | base64)",
+      "email": "${ARTIFACTORY_EMAIL}"
+    }
+  }
+}
+EOF
+fi
+
+# Install the JFrog vscode extension for code-server.
+if [ "${CONFIGURE_CODE_SERVER}" == "true" ]; then
+  while ! [ -x /tmp/code-server/bin/code-server ]; do
+    counter=0
+    if [ $counter -eq 30 ]; then
+      echo "Timed out waiting for /tmp/code-server/bin/code-server to be installed."
+      exit 1
+    fi
+    echo "Waiting for /tmp/code-server/bin/code-server to be installed..."
+    sleep 1
+    ((counter++))
+  done
+  echo "📦 Installing JFrog extension..."
+  /tmp/code-server/bin/code-server --install-extension jfrog.jfrog-vscode-extension
+  echo "🥳 JFrog extension installed!"
+else
+  echo "🤔 Skipping JFrog extension installation. Set configure_code_server to true to install the JFrog extension."
+fi
+
+# Configure the JFrog CLI completion
+echo "📦 Configuring JFrog CLI completion..."
+## Get the user's shell
+SHELLNAME=$(grep "^$USER" /etc/passwd | awk -F':' '{print $7}' | awk -F'/' '{print $NF}')
+## Generate the completion script
+jf completion $SHELLNAME --install
+## Add the completion script to the user's shell profile
+if [ "$SHELLNAME" == "bash" ]; then
+  if ! grep -q "# jf CLI shell completion" ~/.bashrc; then
+    echo "" >> ~/.bashrc
+    echo "# jf CLI shell completion" >> ~/.bashrc
+    echo "source $HOME/.jfrog/jfrog_bash_completion" >> ~/.bashrc
```

Comment:
```suggestion
    echo 'source "$HOME/.jfrog/jfrog_bash_completion"' >> ~/.bashrc
```

Two changes, 1) we leave `$HOME` unresolved, this makes the file more portable and 2) the path _may_ contain spaces (paranoia mode).

(Same below, for zsh.)

Reply:
Also, I have noticed that for some docker workspaces, the `.bashrc` and `.zshrc` files do not exist at creation.

Reply:
handled this.